mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -1361,6 +1384,27 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 	desiredSetting := d.Get("settings")
 	var op *sqladmin.Operation
 	var instance *sqladmin.DatabaseInstance
+	// Check if the maintenance version is being updated, because patching maintenance version is an atomic operation and can not be
+	// performed with other fields, we first patch maintenance version before updating the rest of the fields.
+	if v, ok := d.GetOk("maintenance_version"); ok {
```

Comment:
I think we should add another logic here to check if the config has been changed by adding `if d.HasChange("maintenance_version")`